OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs

```diff
@@ -43,13 +43,20 @@ public static IEnumerable<CPos> FrozenUnderFogTiles(Actor a)
 			return Tiles(a.World.Map.Rules, a.Info.Name, a.Info.TraitInfo<BuildingInfo>(), a.Location, true);
 		}
 
-		public static IEnumerable<CPos> UnpathableTiles(string name, BuildingInfo buildingInfo, CPos position)
+		public static IEnumerable<CPos> TargetableTiles(string name, BuildingInfo buildingInfo, CPos position)
 		{
 			var footprint = buildingInfo.Footprint.Where(x => !char.IsWhiteSpace(x)).ToArray();
 			foreach (var tile in TilesWhere(name, buildingInfo.Dimensions, footprint, a => a == 'x'))
 				yield return tile + position;
 		}
 
+		public static IEnumerable<CPos> UnpathableTiles(string name, BuildingInfo buildingInfo, CPos position)
+		{
+			var footprint = buildingInfo.Footprint.Where(x => !char.IsWhiteSpace(x)).ToArray();
```

Comment:
```csharp
public enum FootprintCellType { Passable, Blocking, ... }
static readonly Dictionary<char, FootprintCellType> FootprintCellTypes = new Dictionary<char, FootprintCellType>()
{
	{ '_', FootprintCellType.Passable },
	{ 'x', FootprintCellType.Blocking },
	...
};
```
```csharp
[FieldLoader.LoadUsing("LoadFootprint")]
public readonly Dictionary<CVec, FootprintCellType> Footprint;

protected static object LoadFootprint(MiniYaml yaml)
{
	var footprintYaml = yaml.Nodes.FirstOrDefault(n => n.Key == "Footprint");
	var footprintChars = footprintYaml != null ? footprintYaml.Value.Value.Where(x => !char.IsWhiteSpace(x)).ToArray() : new[] { 'x' };

	var dimensionsYaml = yaml.Nodes.FirstOrDefault(n => n.Key == "Dimensions");
	var dim = dimensionsYaml != null ? FieldLoader.GetValue<CVec>("Dimensions", dimensionsYaml.Value.Value) : new CVec(1, 1);

	var index = 0;
	var ret = new Dictionary<CVec, FootprintCellType>();
	for (var y = 0; y < dim.Y; y++)
	{
		for (var x = 0; x < dim.X; x++)
		{
			FootprintCellType cell;
			if (!FootprintCellTypes.TryGetValue(footprintChars[index++], out cell))
				throw new YamlException("Invalid footprint cell type '{0}'".F(footprintChars[index - 1]));

			ret[new CVec(x, y)] = cell;
		}
	}

	if (index != dim.X * dim.Y)
		throw new YamlException("Invalid footprint");

	return ret;
}

public IEnumerable<CPos> UnpathableTiles(CPos location)
{
	return NewFootprint.Where(kv => kv.Value == BuildingInfo.FootprintCellType.Blocking).Select(kv => location + kv.Key);
}

...
```
and then you can remove `FootprintUtils` and query directly from the `BuildingInfo`.